manifests/service/puppetdb.pp

```diff
@@ -12,166 +12,327 @@
+      'mbean' => 'puppetlabs.puppetdb.storage:name=gc-fact-paths'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_gc-params-time',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=gc-params-time'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_gc-report-statuses',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=gc-report-statuses'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_gc-time',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=gc-time'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_new-catalog-time',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=new-catalog-time'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_new-catalogs',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=new-catalogs'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_replace-catalog-time',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=replace-catalog-time'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_replace-facts-time',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=replace-facts-time'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_resource-hashes',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=resource-hashes'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_store-report-time',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=store-report-time'
+    }
   ]
 
   # TODO: Track these on a less frequent cadence because they are slow to run
 
   $storage_metrics_db_queries = [
-    { 'name' => 'storage_catalog-volitilty',
-      'url'  => 'puppetlabs.puppetdb.storage:name=catalog-volitilty' },
-    { 'name' => 'storage_duplicate-catalogs',
-      'url'  => 'puppetlabs.puppetdb.storage:name=duplicate-catalogs' },
-    { 'name' => 'storage_duplicate-pct',
-      'url'  => 'puppetlabs.puppetdb.storage:name=duplicate-pct' },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_catalog-volitilty',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=catalog-volitilty'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_duplicate-catalogs',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=duplicate-catalogs'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'storage_duplicate-pct',
+      'mbean' => 'puppetlabs.puppetdb.storage:name=duplicate-pct'
+    }
   ]
 
+  # Retaining for future reference.
   $numbers = $::pe_server_version ? {
-    /^2015.2/     => {'catalogs' => 6, 'facts' => 4, 'reports' => 6},
-    /^2015.3/     => {'catalogs' => 7, 'facts' => 4, 'reports' => 6},
-    /^2016.(1|2)/ => {'catalogs' => 8, 'facts' => 4, 'reports' => 7},
-    /^2016.(4|5)/ => {'catalogs' => 9, 'facts' => 5, 'reports' => 8},
-    /^2017.(1|2)/ => {'catalogs' => 9, 'facts' => 5, 'reports' => 8},
-    default       => {'catalogs' => 9, 'facts' => 5, 'reports' => 8},
+    /^2016\.[45]\./ => {'catalogs' => 9, 'facts' => 5, 'reports' => 8},
```

Comment:
We should cut 2016.4 and 2016.5 as well. They don't have `metrics/v2` and no longer match the version requirement of `puppet >= 5.5.1`.

Reply:
Done!